src/Cli/ConfigGenerator.cs

```diff
@@ -1614,251 +1645,174 @@ public static bool TryUpdateExistingEntity(UpdateOptions options, RuntimeConfig
-            Dictionary<string, EntityRelationship>? updatedRelationships = entity.Relationships;
-            Dictionary<string, string>? updatedMappings = entity.Mappings;
-            EntityActionPolicy? updatedPolicy = GetPolicyForOperation(options.PolicyRequest, options.PolicyDatabase);
-            EntityActionFields? updatedFields = GetFieldsForOperation(options.FieldsToInclude, options.FieldsToExclude);
-            EntityCacheOptions? updatedCacheOptions = ConstructCacheOptions(options.CacheEnabled, options.CacheTtl);
+            EntityMcpOptions? updatedMcpOptions = ConstructUpdatedMcpOptions(entity, options);
 
-            if (!updatedGraphQLDetails.Enabled)
+            // Add these missing variable definitions
+            EntityActionPolicy? policy = GetPolicyForOperation(options.PolicyRequest, options.PolicyDatabase);
+            EntityActionFields? field = GetFieldsForOperation(options.FieldsToInclude, options.FieldsToExclude);
+
+            // Validate that permissions are provided when using field include/exclude
+            if (field is not null && (options.Permissions is null || !options.Permissions.Any()))
             {
-                _logger.LogWarning("Disabling GraphQL for this entity will restrict its usage in relationships");
+                _logger.LogError("Permissions must be specified when using --fields.include or --fields.exclude options.");
+                return false;
             }
 
-            EntitySourceType? updatedSourceType = updatedSource.Type;
+            EntityPermission[]? updatedPermissions;
 
             if (options.Permissions is not null && options.Permissions.Any())
             {
-                // Get the Updated Permission Settings
-                updatedPermissions = GetUpdatedPermissionSettings(entity, options.Permissions, updatedPolicy, updatedFields, updatedSourceType);
+                updatedPermissions = GetUpdatedPermissionSettings(entity, options.Permissions, policy, field, updatedSource.Type);
 
+                // Only fail if we tried to update permissions and it failed
                 if (updatedPermissions is null)
                 {
-                    _logger.LogError("Failed to update permissions.");
+                    _logger.LogError("Failed to update entity permissions.");
                     return false;
                 }
             }
             else
             {
+                updatedPermissions = entity.Permissions;
+            }
 
-                if (options.FieldsToInclude is not null && options.FieldsToInclude.Any()
-                    || options.FieldsToExclude is not null && options.FieldsToExclude.Any())
-                {
-                    _logger.LogInformation("--permissions is mandatory with --fields.include and --fields.exclude.");
-                    return false;
-                }
+            // Handle relationships
+            Dictionary<string, EntityRelationship>? updatedRelationships = null;
```

Comment:
I dont understand why was any of this refactored. How is this related to the updatedMcpOptions change? This could have been avoided to make it easier for PR reviewer.